micro-memcached/src/main/java/com/oath/micro/server/memcached/DistributedCacheFactory.java

```diff
@@ -0,0 +1,43 @@
+package com.oath.micro.server.memcached;
+
+import lombok.extern.slf4j.Slf4j;
+import net.spy.memcached.MemcachedClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+@Component
+@Slf4j
+public class DistributedCacheFactory {
+
+    private final ElasticacheConfig config;
+
+    @Autowired
+    public DistributedCacheFactory(ElasticacheConfig config) {
+        this.config = config;
+    }
+
+    public <K, V> DistributedCache<K, V> create() {
+        try {
+            log.info("Creating Memcached Data connection for elasticache cluster: {}", config.getHostname());
+            return new MemcachedCacheImpl(createMemcachedClient(), config.getRetryAfterSecs(), config.getMaxRetries());
```

Comment:
I guess
```java
return new MemcachedCacheImpl<K, V>(createMemcachedClient(), ...
```

Reply:
IIRC the compiler can deduce the types <K, V> from the return type.

Reply:
besides, the IDE suggests it's not relevant

Reply:
cool, thanks.